k4FWCore/scripts/k4run

```diff
@@ -18,35 +20,83 @@ __filterGaudiProps = [ "ContextService", "Cardinality", "Context", "CounterList"
+        exec(open(wrapper.name).read(), globals())
+      # loop over all components that were configured in the options file
+      # use dict.fromkeys so that the confs are sorted (Python >= 3.7)
+      for conf in dict.fromkeys(ApplicationMgr.allConfigurables.values()):
+        # skip public tools and the applicationmgr itself
+        if "ToolSvc" in conf.name() or "ApplicationMgr" in conf.name():
+          continue
+        props = conf.getPropertiesWithDescription() #dict propertyname: (propertyvalue, propertydescription)
+        for prop in props:
+          # only add arguments for relevant properties
+          if prop in FILTER_GAUDI_PROPS or "Audit" in prop or hasattr(props[prop][0], '__slots__'):
+            continue
+          propvalue = props[prop][0]
+
+          # if it is set to "no value" it hasn't been touched in the options file
+          if propvalue == conf.propertyNoValue:
+              propvalue = conf.getDefaultProperty(prop) # thus get the default value
+          proptype = type(props[prop][0])
+          # if the property is a list of something, we need to set argparse nargs to '+'
+          propnargs = "?"
+          if proptype == list:
+              # tricky edgecase: if the default is an empty list there is no way to get the type
+              if len(propvalue) == 0:
+              # just skip for now
+              #print("Warning: argparse cannot deduce type for property %s of %s. Needs to be set in options file." % (prop, conf.name()))
+                continue
+              else:
+                # deduce type from first item of the list
+                proptype = type(propvalue[0])
+              propnargs = "+"
+
+          # add the argument twice, once as "--PodioOutput.filename"
+          # and once as "--filename.PodioOutput"
+          propName = conf.name() + '.' + prop
+          propNameReversed = prop + '.' + conf.name()
+          option_db[propName] = (conf, propName)
+          parser.add_argument( "--%s" % propName, "--%s" % propNameReversed, type=proptype, help=props[prop][1],
+              nargs=propnargs,
+              default=propvalue)
+
+
 if __name__ == "__main__":
     # ensure that we (and the subprocesses) use the C standard localization
-    if os.environ.get('LC_ALL') != 'C':
-        os.environ['LC_ALL'] = 'C'
-
-    # run all the configuration files
-    # we generate arguments from them, so we need to load them
-    # before parsing the arguments
-    # to avoid parsing them twice, just assume the positional arguments
-    # come first and as soon as we encounter "-" we stop
-    run_with_options_file = False
-    for f in sys.argv[1:]:
-      if f[0] != '-':
-        exec(open(f).read())
-        run_with_options_file = True
-      else:
-        break
+    os.environ['LC_ALL'] = 'C'
+
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    # Don't propagate to the root logger, otherwise the output will be doubled
```

Comment:
At what point is the output doubled? If there is already output that means the logger is already created somewhere else, and we don't have to do it again, right?

Reply:
If I remember correctly, if not changed the logging output goes to stderr (and not stdout), where does the output go to at the moment?

Reply:
Under what circumstances did you test things when removing the InstallRootLoggingHandler?
What happens if you move that call here instead of the logging setup you are adding?